packages/flutter/test/widgets/actions_test.dart

```diff
@@ -1025,6 +1025,676 @@ void main() {
+                            builder: (BuildContext context4) {
+                              invokingContext = context4;
+                              return const SizedBox();
+                            },
+                          ),
+                        );
+                      },
+                    ),
+                  );
+                },
+              ),
+            );
+          },
+        ),
+      );
+
+      Actions.invoke(invokingContext!, LogIntent(log: invocations));
+      expect(invocations, <String>[
+        'action1.invokeAsOverride-pre-super',
+        'action2.invokeAsOverride-pre-super',
+        'action3.invoke',
+        'action2.invokeAsOverride-post-super',
+        'action1.invokeAsOverride-post-super',
+      ]);
+    });
+
+    testWidgets('Overriding Actions can change the intent', (WidgetTester tester) async {
+      final List<String> newLogChannel = <String>[];
+      await tester.pumpWidget(
+        Builder(
+          builder: (BuildContext context1) {
+            return Actions(
+              actions: <Type, Action<Intent>> {
+                LogIntent: Action<LogIntent>.overridable(defaultAction: LogInvocationAction(actionName: 'action1'), context: context1),
+              },
+              child: Builder(
+                builder: (BuildContext context2) {
+                  return Actions(
+                    actions: <Type, Action<Intent>> {
+                      LogIntent: Action<LogIntent>.overridable(defaultAction: RedirectOutputAction(actionName: 'action2', newLog: newLogChannel), context: context2),
+                    },
+                    child: Builder(
+                      builder: (BuildContext context3) {
+                        return Actions(
+                          actions: <Type, Action<Intent>> {
+                            LogIntent: Action<LogIntent>.overridable(defaultAction: LogInvocationAction(actionName: 'action3'), context: context3),
+                          },
+                          child: Builder(
+                            builder: (BuildContext context4) {
+                              invokingContext = context4;
+                              return const SizedBox();
+                            },
+                          ),
+                        );
+                      },
+                    ),
+                  );
+                },
+              ),
+            );
+          },
+        ),
+      );
+
+      Actions.invoke(invokingContext!, LogIntent(log: invocations));
+      expect(invocations, <String>[
+        'action1.invokeAsOverride-pre-super',
+        'action1.invokeAsOverride-post-super',
+      ]);
+      expect(newLogChannel, <String>[
+        'action2.invokeAsOverride-pre-super',
+        'action3.invoke',
+        'action2.invokeAsOverride-post-super',
+      ]);
+    });
+
+    testWidgets('Override regular Action with a ContextAction', (WidgetTester tester) async {
```

Comment:
What do you mean by "regular" action here?

I was wondering about the interaction between Action.overridable and plain Action, but that doesn't seem to be this test (or the next one).  What do you think about adding a test where one of these is just an Action (not .overridable)?

Reply:
That case should already be covered by the "Does not override if not overridable" test. This test is about overriding an overridable non-context Action with a `ContextAction`. When invoking the `ContextAction` the correct `BuildContext` should be supplied. Updated the name.